### PR TITLE
Always use Docs URL to check redirection rule with document_id as true

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#204994](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/204994)

#5806 TODO-1

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5826)

### CODE REVIEW

File: src/docfx/build/document/DocumentProvider.cs [hunk 1]
```diff
@@ -62,6 +62,17 @@ public string GetOutputPath(FilePath path)
             return UrlUtility.Combine(_config.BasePath, outputPath);
         }
 
+        public string GetDocsSiteUrl(FilePath path, OutputUrlType outputUrlType)
```
Comment (yufeih) [inline src/docfx/build/document/DocumentProvider.cs:65]: If this already takes `OutputUrlType`, consider remove `Docs` from the name
```suggestion
        public string GetSiteUrl(FilePath path, OutputUrlType outputUrlType)
```

File: src/docfx/build/redirection/RedirectionProvider.cs [hunk 2]
```diff
@@ -218,11 +217,11 @@ private static IEnumerable<string> ProbeRedirectionFiles(string docsetPath, Repo
             return renameHistory;
         }
 
-        private Dictionary<string, List<FilePath>> GetPublishUrlMap(IEnumerable<FilePath> redirectUrlSources)
+        private Dictionary<string, List<FilePath>> GetPublishUrlMap(IEnumerable<FilePath> redirectUrlSources, OutputUrlType outputUrlType)
         {
             var fileUrls = new ConcurrentBag<(FilePath file, string url)>();
             var allSources = _buildScope.Files.Concat(redirectUrlSources);
-            ParallelUtility.ForEach(_errorLog, allSources, file => fileUrls.Add((file, _documentProvider.GetDocument(file).SiteUrl)));
+            ParallelUtility.ForEach(_errorLog, allSources, file => fileUrls.Add((file, _documentProvider.GetDocsSiteUrl(file, outputUrlType))));
```
Comment (yufeih) [inline src/docfx/build/redirection/RedirectionProvider.cs:224]: Is this always `docs`?
```suggestion
            ParallelUtility.ForEach(_errorLog, allSources, file => fileUrls.Add((file, _documentProvider.GetDocsSiteUrl(file, OutputUrlType.Docs))));
```

File: src/docfx/build/document/DocumentProvider.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/build/document/DocumentProvider.cs:65]: The parameter `outputUrlType` means the current config outputUrlType, not the target urlType

---
In reply to: [415203921](https://github.com/dotnet/docfx/pull/5826#discussion_r415203921) [](ancestors = 415203921)

File: src/docfx/build/redirection/RedirectionProvider.cs (hunk 2 above)
Comment (928PJY) [reply src/docfx/build/redirection/RedirectionProvider.cs:224]: No, this parameter means the current config OutputUrlType, not the target urlType.

---
In reply to: [415204204](https://github.com/dotnet/docfx/pull/5826#discussion_r415204204) [](ancestors = 415204204)

File: docs/specs/static.yml [hunk 3]
```diff
@@ -64,6 +64,33 @@ inputs:
 outputs:
   a.html:
 ---
+# Always ues Docs URL to check redirection rule with document_id as true
```
Comment (yufeih) [inline docs/specs/static.yml:67]:
```suggestion
# Always use Docs URL to check redirection rule with document_id as true
```

**Pull request merged**